pkg/remoteimageasync/synchronizer_test.go

```diff
@@ -0,0 +1,281 @@
+package remoteimageasync
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type chanTestStruct struct {
+	str string
+	err error
+}
+
+// demonstrates session channel structure's pass-by-reference is appropriate
+func TestChannelStructContent(t *testing.T) {
```

Comment:
Same comment as above. If these tests are not directly or indirectly related by code to synchronizer, I think we should put it in a separate file.